From: "Aneesh Kumar K.V" <aneesh.kumar@linux.ibm.com>
Subject: powerpc-book3s64-radix-add-support-for-vmemmap-optimization-for-radix-fix
Date: Wed, 26 Jul 2023 10:52:33 +0530

We should look at ppc64 not including radix headers if
CONFIG_PPC_RADIX_MMU is disabled.

Link: https://lkml.kernel.org/r/87zg3jw8km.fsf@linux.ibm.com
Signed-off-by: "Aneesh Kumar K.V" <aneesh.kumar@linux.ibm.com>
Reported-by: Stephen Rothwell <sfr@canb.auug.org.au>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/powerpc/include/asm/book3s/64/radix.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/arch/powerpc/include/asm/book3s/64/radix.h~powerpc-book3s64-radix-add-support-for-vmemmap-optimization-for-radix-fix
+++ a/arch/powerpc/include/asm/book3s/64/radix.h
@@ -364,8 +364,10 @@ int radix__remove_section_mapping(unsign
 
 void radix__kernel_map_pages(struct page *page, int numpages, int enable);
 
+#ifdef CONFIG_ARCH_WANT_OPTIMIZE_DAX_VMEMMAP
 #define vmemmap_can_optimize vmemmap_can_optimize
 bool vmemmap_can_optimize(struct vmem_altmap *altmap, struct dev_pagemap *pgmap);
+#endif
 
 #define vmemmap_populate_compound_pages vmemmap_populate_compound_pages
 int __meminit vmemmap_populate_compound_pages(unsigned long start_pfn,
_
